src/types.rs

```diff
@@ -165,12 +169,13 @@
 
 pub type SessionId = String;
 pub type LanguageId = String;
+pub type ServerName = String;
 pub type RootPath = String;
 
 #[derive(PartialEq, Eq, Hash, Debug, Clone)]
 pub struct Route {
     pub session: SessionId,
-    pub language: LanguageId,
+    pub server_name: ServerName,
```

Comment:
(nit: the removal of `language` should happen in this commit (it's replaced by `server_name`))